pkg/frontend/mysql_cmd_executor.go

```diff
@@ -2143,6 +2161,93 @@ func (cwft *TxnComputationWrapper) Compile(requestCtx context.Context, u interfa
 	if err != nil {
 		return nil, err
 	}
+	// check if it is necessary to initialize the temporary engine
+	if cwft.compile.NeedInitTempEngine(cwft.ses.IfNeedInitTempEngine()) {
+		// 0. init memory-non-dist engine
+		ck := clock.DefaultClock()
+		if ck == nil {
+			ck = clock.NewHLCClock(func() int64 {
+				return time.Now().Unix()
+			}, math.MaxInt)
+		}
+		shard := logservicepb.DNShardInfo{
+			ShardID:   2,
```

Comment:
What are these Shard/ReplicaId?   Why 2?

Reply:
These parameters are fictitious and are not used for the memory citation engine.

Reply:
> What are these Shard/ReplicaId? Why 2?

This is referenced to the initMemoryEngineNoDist function.